### PR TITLE
Add documentation for remote debugging of integration tests.

### PR DESCRIPTION
**Pull request by jmurret**
### Description

Adding documentation for the remote debugging functionality that was added to our integration tests in https://github.com/hashicorp/consul/pull/16887

### CODE REVIEW

File: test/integration/consul-container/test/debugging.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	  -  [Remote debugging using GoLand](#remote-debugging-using-goland)
+
+
+## Introduction
+
+Remote debugging integration tests allows you to attach your debugger to the consul container and debug go code running on that container. 
+
+### How it works
+The `dev-docker-dbg` Make target will build consul docker container that has the following:
+- [delver (dlv) debugger](https://github.com/go-delve/delve) installed.
```
Comment (dhiaayachi) [inline test/integration/consul-container/test/debugging.md:18]:
```suggestion
- [delve (dlv) debugger](https://github.com/go-delve/delve) installed.
```

**Review by dhiaayachi — APPROVED** [review]
This is great @jmurret 👏 👏 👏 !!

I added few nits but otherwise ✅

File: test/integration/consul-container/test/debugging.md (hunk 1 above)
Comment (jmurret) [reply test/integration/consul-container/test/debugging.md]: Thank you!  Included in 34bd61e44f2b3de4deec9e8d10c9df4ef1e1c528

File: test/integration/consul-container/test/debugging.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  ```
+  make dev-docker-dbg
+  ```
+- Build a consul-envoy container image from the consul root directory that is required for testing but not for debugging. 
+  ```
+  docker build consul-envoy:target-version --build-arg CONSUL_IMAGE=consul:local --build-arg ENVOY_VERSION=1.24.6 -f ./test/integration/consul-container/assets/Dockerfile-consul-envoy ./test/integration/consul-container/assets
+  ```
+
+#### Remote debugging using GoLand
+(For additional information, see [GoLand's documentation on remote debugging](https://www.jetbrains.com/help/go/attach-to-running-go-processes-with-debugger.html#attach-to-a-process-on-a-remote-machine).)
```
[the remark this answers is not in the document]
Comment (jmurret) [reply test/integration/consul-container/test/debugging.md:45]: Good call.  THis is included in 34bd61e44f2b3de4deec9e8d10c9df4ef1e1c528

**Pull request merged**